### PR TITLE
Add Amazon Provider Package Fix to Local Runner

### PR DESCRIPTION
**Pull request by john-jac**
*Issue #, if available:*
175

*Description of changes:*
This aligns local runner to the recent improvements in MWAA that support specifying newer Amazon provider package versions.

Using this same technique, other core packages may also be updated.

See https://docs.aws.amazon.com/mwaa/latest/userguide/connections-packages.html#connections-packages-newer-packages

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docker/script/bootstrap.sh [hunk 1]
```diff
@@ -1,51 +1,65 @@
 #!/bin/sh
 
 set -e
+set -x
```
Comment (o-nikolas) [inline docker/script/bootstrap.sh:4]: Did you intend to leave this in? The logging that comes from this is great for debug but can be overwhelming for prod applications.

File: docker/script/entrypoint.sh [hunk 2]
```diff
@@ -23,6 +23,19 @@ export \
...
+      then
+          # --constraint found
+          :
+      else
+          if grep -- "--constraint " "$AIRFLOW_HOME/$REQUIREMENTS_FILE"
+          then
+              # --constraint found
+              :
+          else
+              echo "WARNING: Constraints should be specified for requirements.txt. Please see https://docs.aws.amazon.com/mwaa/latest/userguide/working-dags-dependencies.html#working-dags-dependencies-test-create"
```
Comment (o-nikolas) [inline docker/script/entrypoint.sh]: Why not negate this and use an and operator to get rid of the empty branches?

File: docker/Dockerfile [hunk 3]
```diff
@@ -19,6 +19,7 @@ ARG PYTHON_DEPS=""
 ARG SYSTEM_DEPS=""
 ARG INDEX_URL=""
 ENV AIRFLOW_HOME=${AIRFLOW_USER_HOME}
+ENV PATH="/usr/local/airflow/.local/bin:/root/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/airflow/.local/lib/python3.7/site-packages:/usr/local/airflow/.local/bin:/usr/local/airflow/.local/lib/python3.7/site-packages:/usr/local/airflow/.local/bin:/usr/local/airflow/.local/lib/python3.7/site-packages"
```
Comment (o-nikolas) [inline docker/Dockerfile]: There are more paths here than I would have expected.

You're also completely overwriting `PATH`. Usually you'd just update/append to it with the specific paths you want to add:
`ENV PATH="$PATH:/usr/local/airflow/.local/lib/python3.7/site-packages:..."`
I think some of the paths you're adding here should be in PATH by default:

File: dags/example_lambda.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# under the License.
+
+import json
+from datetime import datetime, timedelta
+from os import getenv
+
+from airflow import DAG
+from airflow.providers.amazon.aws.operators.aws_lambda import AwsLambdaInvokeFunctionOperator
+
+# [START howto_operator_lambda_env_variables]
```
Comment (o-nikolas) [inline dags/example_lambda.py]: These START/END comments are used automatically by Airflow docs, but won't be used here. I'd drop them for simplicity.

File: docker/script/bootstrap.sh (hunk 1 above)
Comment (john-jac) [reply docker/script/bootstrap.sh:4]: Ideally they're using local-runner for debugging and not prod, and this reflects what we have in MWAA

File: docker/script/entrypoint.sh (hunk 2 above)
Comment (john-jac) [reply docker/script/entrypoint.sh]: Done

File: docker/Dockerfile (hunk 3 above)
Comment (john-jac) [reply docker/Dockerfile]: Done

File: dags/example_lambda.py (hunk 4 above)
Comment (john-jac) [reply dags/example_lambda.py]: Done

File: docker/script/entrypoint.sh [hunk 5]
```diff
@@ -23,6 +23,13 @@ export \
 install_requirements() {
     # Install custom python package if requirements.txt is present
     if [[ -e "$AIRFLOW_HOME/$REQUIREMENTS_FILE" ]]; then
+      if ! grep -- "-c " "$AIRFLOW_HOME/$REQUIREMENTS_FILE"
+      then
+          if ! grep -- "--constraint " "$AIRFLOW_HOME/$REQUIREMENTS_FILE"
```
Comment (o-nikolas) [inline docker/script/entrypoint.sh:28]: Nit: You could also combine these with an OR (using an escaped pipe) something like: `grep 'foo\|bar' /path/to/file`).

But it's a minor improvement, feel free to make it or not, approving :+1:

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**